File: src/Backend/Sudoku/Application/Datasets/CreateDatasetPreparationCommandResultDto.cs
namespace Sudoku.Application.Datasets;

public sealed record CreateDatasetPreparationCommandResultDto(
    string PreparationName,
    DateTimeOffset CreatedAtUtc,
    string Status,
    IReadOnlyList<DatasetPreparationSourceReportDto> Sources,
    IReadOnlyList<string> Warnings);

File: src/MachineLearning/application/features/datasets/commands/create_dataset_preparation/create_dataset_preparation_command_result_dto.py
from dataclasses import dataclass

from application.features.datasets.dto.dataset_preparation_source_report_dto import (
    DatasetPreparationSourceReportDto,
)


@dataclass(frozen=True)
class CreateDatasetPreparationCommandResultDto:
    preparation_name: str
    created_at_utc: str
    status: str
    source_reports: tuple[DatasetPreparationSourceReportDto, ...]
    warnings: tuple[str, ...]
